generators/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/interceptor/EventListenerInterceptorChainProviderCodeGenerator.java

```diff
@@ -0,0 +1,126 @@
+package uk.gov.justice.raml.jms.interceptor;
+
+import static com.squareup.javapoet.ClassName.get;
+import static com.squareup.javapoet.MethodSpec.constructorBuilder;
+import static com.squareup.javapoet.MethodSpec.methodBuilder;
+import static com.squareup.javapoet.TypeSpec.classBuilder;
+import static javax.lang.model.element.Modifier.FINAL;
+import static javax.lang.model.element.Modifier.PRIVATE;
+import static javax.lang.model.element.Modifier.PUBLIC;
+
+import uk.gov.justice.services.components.event.listener.interceptors.EventBufferInterceptor;
+import uk.gov.justice.services.core.interceptor.InterceptorChainEntry;
+import uk.gov.justice.services.core.interceptor.InterceptorChainEntryProvider;
+
+import java.util.List;
+
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+
+/**
+ * Should generate a custom EventListenerInterceptorChainProvider that uses a custom generated
+ * EventFilterInterceptor.
+ *
+ * Something like this:
+ *
+ * <pre>
+ *  {@code
+ *
+ *     public class MyCustomEventListenerInterceptorChainProvider implements InterceptorChainEntryProvider {
+ *
+ *          private final List<InterceptorChainEntry> interceptorChainEntries = new ArrayList<>();
+ *
+ *          public ExampleEventListenerInterceptorChainProvider() {
+ *              interceptorChainEntries.add(new InterceptorChainEntry(1000, EventBufferInterceptor.class));
+ *              interceptorChainEntries.add(new InterceptorChainEntry(2000, MyCustomEventFilterInterceptor.class));
+ *          }
+ *
+ *          @literal @Override
+ *          public String component() {
+ *              return "MY_CUSTOM_EVENT_LISTENER";
+ *          }
+ *
+ *          @literal @Override
+ *          public List<InterceptorChainEntry> interceptorChainTypes() {
+ *              return interceptorChainEntries;
+ *          }
+ *     }
+ *
+ * }
+ * </pre>
+ */
+public class EventListenerInterceptorChainProviderCodeGenerator {
```

Comment:
Does this still work with people wanting to create / append their own providers?

Such as the authorisation interceptor and things?

Reply:
People are still able to add into the Interceptor Chain Provider by implementing another InterceptorChainEntryProvider....both providers are loaded from the classpath and combined at start up...they will not be able to remove from the chain though

Reply:
ok cool, I assumed so but just wanted to check.